csm_web/scheduler/views/user.py

```diff
@@ -23,12 +23,29 @@ def list(self, request):
         return Response(self.queryset.order_by("email").values_list("email", flat=True))
 
 
-@api_view(["GET"])
+@api_view(["GET", "PUT"])
 def userinfo(request):
     """
     Get user info for request user
 
     TODO: perhaps replace this with a viewset when we establish profiles
     """
-    serializer = UserSerializer(request.user)
-    return Response(serializer.data, status=status.HTTP_200_OK)
+    if request.method == "GET":
+        serializer = UserSerializer(request.user)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    elif request.method == "PUT":
+        user = User.objects.get(id=request.user)
+        bio = request.data.get("bio")
+        pronunciation = request.data.get("pronunciation")
+        pronouns = request.data.get("pronouns")
+        is_private = request.data.get("is_private")
+        if bio is not None:
+            user.bio = bio
+        if pronunciation is not None:
+            user.pronunciation = pronunciation
+        if pronouns is not None:
+            user.pronouns = pronouns
+        if is_private is not None:
+            user.is_private = is_private
+        user.save()
+        return Response(status=status.HTTP_201_CREATED)
```

Comment:
Ideally, this shouldn't be a `201 CREATED`, since you haven't actually created anything. Likely a `200 OK` or a ``202 ACCEPTED` would likely fit better.

Reply:
yes sir